### PR TITLE
Write metafile for TPC residuals

### PR DESCRIPTION
**Pull request by martenole**
And make output directory for residuals configurable

At the moment this is WIP. Will add the TF statistics information in an additional commit.

### CODE REVIEW

**Comment by martenole** [comment]
Waiting with this until #8533 is merged which will give me easy access to the TF ID information

**Comment by martenole** [comment]
Hi @shahor02 
this would be ready for review now.
Cheers,
Ole

File: Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx [hunk 1]
```diff
@@ -182,7 +205,7 @@ bool ResidualAggregator::hasEnoughData(const Slot& slot) const
 
 void ResidualAggregator::initOutput()
 {
-  // nothing to be done here, but method needs to be overwritten
+  mFileMetaData = std::make_unique<o2::dataformats::FileMetaData>();
```
Comment (shahor02) [inline Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx]: Why do you need this? It is anyway assigned at L227. You could actually use a local FileMetaData object at that line, w/o any smart pointers. I will also change this in the CTFWriter.

File: Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx [hunk 2]
```diff
@@ -195,15 +218,41 @@ void ResidualAggregator::finalizeSlot(Slot& slot)
...
   cont->treeOutStats.reset();
+  cont->treeOutRecords->Write();
+  cont->treeOutRecords.reset();
   cont->fileOut->Close();
   cont->fileOut.reset();
+  std::filesystem::rename(o2::utils::Str::concat_string(mOutputDir, cont->fileName, ".part"), mOutputDir + cont->fileName);
+  if (mStoreMetaData) {
+    mFileMetaData = std::make_unique<o2::dataformats::FileMetaData>();
+    mFileMetaData->fillFileData(mOutputDir + cont->fileName);
+    // mFileMetaData->run = getTFrunNumber();
```
Comment (shahor02) [inline Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx]: Why did you comment setting the run number?

File: Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx (hunk 1 above)
Comment (martenole) [reply Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx]: Sorry I did not really think about it when I took that from the CTF writer.. It's changed now to a local object

File: Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx (hunk 2 above)
Comment (martenole) [reply Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx]: I actually forgot to fix this, that is why. I am taking the run number now from the TimeSlot which in turn gets it through the TFIDInfo. Thanks for spotting!

**Comment by martenole** [comment]
Hi @shahor02 do you think this could be merged?
Cheers, Ole

**Review by shahor02 — APPROVED** [review]

**Pull request merged**